src/net.cpp

```diff
@@ -1945,9 +1919,45 @@ void static Discover(boost::thread_group& threadGroup)
 #endif
 }
 
-void StartNode(boost::thread_group& threadGroup, CScheduler& scheduler)
+CConnman::CConnman()
+{
+    setBannedIsDirty = false;
+    fAddressesInitialized = false;
+    nLastNodeId = 0;
+    nSendBufferMaxSize = 0;
+    nReceiveFloodSize = 0;
+    semOutbound = NULL;
+    nMaxConnections = 0;
+    nMaxOutbound = 0;
+    nBestHeight = 0;
+    clientInterface = NULL;
+}
+
+NodeId CConnman::GetNewNodeId()
+{
+    return nLastNodeId.fetch_add(1, std::memory_order_relaxed);
+}
+
+bool CConnman::Start(boost::thread_group& threadGroup, CScheduler& scheduler, std::string& strNodeError, Options connOptions)
```

Comment:
Roger. I see your point and agree, I'm just avoid overthinking this until the async stuff is added. Once it's all hooked up, maybe we can switch to a model where Start() returns some kind of object that guarantees a certain state that CConnman does not.

The alternative would be to move out things that don't require a running state (addrman/bandb as examples) and let them have separate life-cycles, but I'd like to do that as a next step once all callers understand that those things aren't global.